app/models/manageiq/providers/openstack/infra_manager/host/operations.rb

```diff
@@ -0,0 +1,24 @@
+module ManageIQ::Providers::Openstack::InfraManager::Host::Operations
+  def self.included(base)
+    base.send :include, InstanceMethods
+    base.extend ClassMethods
+  end
+
+  module InstanceMethods
+    def ironic_fog_node
+      fog_baremetal = ext_management_system.openstack_handle.baremetal_service
+      fog_baremetal.nodes.find_by_uuid(uid_ems)
```

Comment:
not a bug, get_node is a request returning just http response, nodes.get is a model call returning model

please use the 

connection_options = {:service => "Baremetal"}
ext_management_system.with_provider_connection(connection_options) do |service|
  servicel.nodes.get(uid_ems)
end